torch/ao/pruning/_experimental/pruner/README.md

```diff
@@ -1,93 +1,263 @@
+```
 
-The Base Pruner inherits from the Base Sparsifier.
+Let's walk through this line by line to understand what the different parts are doing.
 
+###  Pruning Config
 
-# Motivation
+To specify the layers to prune we just need the fully qualified name (FQN) of the tensor you are looking to prune in the module.
+You can get the FQN of a tensor by printing out `model.named_parameters()`.
 
-Sparsifying weights allows us to skip some of the multiplications during the dot product (i.e. in the Linear layers), which ultimately translates into faster inference. With structured pruning, whole rows/columns of a tensor would be zeroed-out. This translates into model transformation (not just tensor transformation). Logically, the process of structured pruning is similar to removing some of the input/output channels in the layer completely.
+For example, for the model defined above, let's say that we want to prune the first Linear layer (Alpha)
 
-![prune logic](./images/prune_1.png)
+To do this we would define a config that looks like this:
 
+``` python
+pruning_config = [
+    {"tensor_fqn": "seq.0.weight"}
+]
+```
 
-# Design Choices
+To prune multiple layers, we just append entries to the pruning config.
+**tensor_fqn** is the only required key in the pruning config. You can pass additional information in the config, for example the sparsity level you want to prune to by adding a key to the config. You can then access this additional information when you update the masks.
 
+``` python
+pruning_config = [
+    {"tensor_fqn": "seq.0.weight", "sparsity_level": 0.7},
+    {"tensor_fqn": "seq.2.weight", "sparsity_level": 0.7},
+    {"tensor_fqn": "seq.4.weight", "sparsity_level": 0.7},
+    {"tensor_fqn": "linear.weight", "sparsity_level": 0.7},
+]
+```
 
-## Eager Mode
+### Implementing a Pruner
 
-**PruningParametrization:** After pruning, the shape of the weight changes (some of the output channels are pruned). That means the output of the current layer will have less output layers compared to the original. This means that the next layer should have less input channels.
+Now that we've defined the sparse config for our model, we need to specify the pruning criteria that we will use.
+To do this, we need to extend a `BaseStructuredSparsifier` with a custom `update_mask` function.
 
-Consider an example below:
+This `update_mask` function contains the user logic for picking what weights to prune.
 
-![prune example](./images/prune_2.png)
+One common pruning criteria is to use the **saliency** of a row, which is defined as the sum of all the L1 norms of the weights in the row.
+The idea is to remove the weights that are small, since they wouldn't contribute much to the final prediction.
 
-The dot product of the masked matrix A (weight) and matrix B (activation) produces the zeros at the sparse locations. However, if we remove the zeros, as in the example shown earlier, the result will change:
+Below we can see an implemented Saliency Pruner
```

Comment:
I think ideally users will not need to implement their own Pruners, I'm not sure if this is necessary for an introduction